### PR TITLE
implement secure staging

### PR DESCRIPTION
**Pull request by nimakaviani**
- break staging into three containers for download, build, upload
- mount certs dir, download appbits and buildpacks
- mount dir for appbits and buildpacks, run buildpack, put droplet in
output dir
- mount certs dir and output dir, upload droplet

[#163723137](https://www.pivotaltracker.com/story/show/163723137)

Signed-off-by: Matthew Broomfield <mattysweeps@google.com>

### CODE REVIEW

File: integration/recipe/staging_test.go [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+		Expect(err).NotTo(HaveOccurred())
+		err = os.Unsetenv(eirini.EnvOutputBuildArtifactsCache)
+		Expect(err).NotTo(HaveOccurred())
+		err = os.Unsetenv(eirini.EnvEiriniAddress)
+		Expect(err).NotTo(HaveOccurred())
+		err = os.Unsetenv(eirini.EnvPacksBuilderPath)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	FContext("when a droplet needs building...", func() {
```
Comment (mattysweeps) [inline integration/recipe/staging_test.go]: Remove focus

**Comment by alex-slynko** [comment]
@nimakaviani @mattysweeps I have pushed vendor clean up separately. 
We will migrate to go mod meanwhile

**Comment by nimakaviani** [comment]
sounds good. we are already gomod-ing in https://github.com/cloudfoundry-incubator/eirini-staging

File: k8s/desiretask.go [hunk 2]
```diff
@@ -13,6 +13,8 @@ import (
 const (
 	ActiveDeadlineSeconds = 900
 	stagingSourceType     = "STG"
+	parallelism           = 1
+	completions           = 3
```
Comment (mnitchev) [inline k8s/desiretask.go]: Does this mean that the job will be run 3 times? Why is this required?

File: models.go [hunk 3]
```diff
@@ -11,20 +11,39 @@ import (
...
+	RecipeBuildPacksDir             = "/var/lib/buildpacks"
+	RecipeBuildPacksName            = "recipe-buildpacks"
+	RecipeWorkspaceDir              = "/recipe_workspace"
+	RecipeWorkspaceName             = "recipe-workspace"
+	RecipeOutputName                = "staging-output"
+	RecipeOutputLocation            = "/out"
+	RecipeOutputDropletLocation     = "/out/droplet.tgz"
+	RecipeOutputBuildArtifactsCache = "/cache/cache.tgz"
+	RecipeOutputMetadataLocation    = "/out/result.json"
+	RecipePacksBuilderPath          = "/packs/builder"
```
Comment (mnitchev) [inline models.go]: This one and `RecipeOutputMetadataLocation`, `RecipeOutputBuildArtifactsCache `, `RecipeOutputDropletLocation` appear to not be used. Can you check whether all of these Environment variable names are used and delete redundant ones.

File: .golangci.yml [hunk 4]
```diff
@@ -36,12 +36,16 @@ issues:
...
       - unused
-    - path: recipe/cmd/buildpack/manager.go
+    - path: recipe/buildpack_manager.go
       linters:
       - gosec
-    - path: integration/recipe/buildpackmanager_test.go
+    - path: recipe/buildpackmanager_test.go
       linters:
       - gosec
+    - path: integration/recipe/staging_test.go
```
[the remark this answers is not in the document]
Comment (mattysweeps) [reply .golangci.yml]: @mnitchev I cleaned up this file. Should be fixed.

File: models.go (hunk 3 above)
Comment (mattysweeps) [reply models.go]: @mnitchev  I unused constants from this file. Should be good.

File: k8s/desiretask.go (hunk 2 above)
Comment (alex-slynko) [reply k8s/desiretask.go]: @mattysweeps this variable requires task to complete three times before it is marked successful. Why is that?

Comment (nimakaviani) [reply k8s/desiretask.go]: It is an oversight from when we were trying to do the staging steps in separate pods in a job. the value should be set to `1` but since parallelism is already at `1`, the value for `completions` has no impact anyway. 

That said, the last table [here](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#job-patterns) describes the relationship between `parallelism` and `completions` in a kubernetes job. It certainly doesnt mean to have the task completed three times.

**Comment by alex-slynko** [comment]
We will fix our CI and then going to merge it
